src/config.rs

```diff
@@ -0,0 +1,281 @@
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::vec::Vec;
+
+use bitcoin::util::bip32::ExtendedPubKey;
+use serde::{de, Deserialize, Deserializer};
+
+/// Everything we need to know for talking to bitcoind serenely
+#[derive(Debug, Deserialize, Clone)]
+pub struct BitcoindConfig {
+    /// The network we are operating on, one of "mainnet", "testnet", "regtest"
+    pub network: String,
+    /// Path to bitcoind's cookie file, to authenticate the RPC connection
+    // TODO: think more about our potential need for the datadir
+    pub cookie_path: PathBuf,
+    /// The IP:port bitcoind's RPC is listening on
+    pub addr: SocketAddr,
+}
+
+/// A participant not taking part in day-to-day fund management, and who runs
+/// a cosigning server to ensure that spending transactions are only signed once.
+#[derive(Debug)]
+pub struct NonManager {
```

Comment:
I am not sure I like the name, maybe `Spender`

Reply:
Some rationale on https://github.com/re-vault/practical-revault/pull/13 and on MM: https://chat.revault.dev:9876/revault/pl/r6twixf65p8f9d5k1r6wpxjf7r